cwms-data-api/src/main/java/cwms/cda/data/dto/stream/StreamLocation.java

```diff
@@ -0,0 +1,171 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including but not limited to the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dto.stream;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.CwmsDTOBase;
+import cwms.cda.data.dto.LocationIdentifier;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV1;
+
+import java.util.Objects;
+
+@FormattableWith(contentType = Formats.JSON, formatter = JsonV1.class)
+@JsonDeserialize(builder = StreamLocation.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class StreamLocation implements CwmsDTOBase {
+
+    private final LocationIdentifier streamLocationId;
+    private final StreamJunctionIdentifier streamJunctionIdentifier;
+    private final Double publishedStation;
+    private final Double navigationStation;
+    private final Double lowestMeasurableStage;
+    private final Double totalDrainageArea;
+    private final Double ungagedDrainageArea;
+
+    private StreamLocation(Builder builder) {
+        this.streamJunctionIdentifier = builder.streamJunctionIdentifier;
+        this.streamLocationId = builder.streamLocationId;
+        this.publishedStation = builder.publishedStation;
+        this.navigationStation = builder.navigationStation;
+        this.lowestMeasurableStage = builder.lowestMeasurableStage;
+        this.totalDrainageArea = builder.totalDrainageArea;
+        this.ungagedDrainageArea = builder.ungagedDrainageArea;
+    }
+
+    @Override
+    public void validate() throws FieldException {
+        if (this.streamLocationId == null) {
```

Comment:
Should also call `streamLocationId.validate()`

Reply:
added validate call